kahuna/public/js/services/image-accessor.js

```diff
@@ -29,6 +29,12 @@ imageAccessor.factory('imageAccessor', function() {
         return image.data.metadata;
     }
 
+    function readExtraInfo(image) {
+        return {
+            filename: image.data.uploadInfo && image.data.uploadInfo.filename
```

Comment:
I've gone with `extraInfo` as we might want something else in there at some stage.